packages/tools/kadena-cli/src/utils/tableDisplay.ts

```diff
@@ -31,11 +34,55 @@ export function displayTable(
         .join(includeVerticalSeparator ? '-+-' : ' ')
     : '';
 
-  const body = rows.map(formatRow).join('\n');
+  // To support multi-line cells, we need to split the cells into multiple rows
+  // and then merge them back together after padding each cell to the correct width.
+  // Rows have multiline in public keys
+  // Example: ['devnet.yaml', 'w:eWE6XbuT....4:keys-any', 'f303a2e0cd....c4b19a9abc\n2105f2f78c....3bf17f621c', 'keys-all', 'coin']
+  // Alias              Account Name             Public Key(s)                                     Predicate Fungible
+  // ------------------ ------------------------ ------------------------------------------------- --------- --------
+  // devnet.yaml        w:eWE6XbuT....4:keys-any f303a2e0cd....c4b19a9abc                          keys-all  coin
+  //                                             2105f2f78c....3bf17f621c
+  const maxColumns = Math.max(...rows.map((row) => row.length));
+
+  const body = rows
+    .map((row) => {
+      const rowStrings = row.map((item, index) => {
+        const cell = item.toString().split('\n');
+        const columnWidth = columnWidths[index];
+        return cell.map((data: string) => data.padEnd(columnWidth));
+      });
+
+      const maxRows = Math.max(
+        ...rowStrings.map((cellData) => cellData.length),
+      );
+
+      //  We need to pad the rows with empty strings to make sure all rows have the same length
+      // so that new line data from the same column aligned properly to the header
+      const contentRows = [];
```

Comment:
i like it, but maybe you can modularise the complexity, by extracting the logic for caculating the column with, and the formatting of the row

calculateColumWidth fn
formatRowfn

since its now became a bit large, and its less readible. 

Other then that nice addition